### DISCUSSION TITLE
How to get a value into a form token?

### DISCUSSION THREAD

**Discussion by AllTheUserNamesAreGone** [filamentphp/filament#12259] [Help]
### Package

Form builder

### Package Version

3

### How can we help you?

Hi clutching at straws trying to get Filament to work and I came across Hidden which can be seen on the site like this:

``` use Filament\Forms\Components\Hidden;
 
Hidden::make('token')```

I have tried creating a hidden TextInput field in order to transfer data from the form into the save method loading it with ```->default('ValueNeedToPassToSave')``` but needless to say that doesn't work when it's hidden.

Anyway I thought I'd try this ```Hidden::`` instead but I can not get a value into in order to get it to the form save() from my Resource file. I see the token 'Category' but is of course null. I have googled it looking for examples  but I find nothing and no example on how to use it on the Filament website either.

It says it needs a 'static' as the value but I have this in my form and have tried initialising with static variables, fn etc etc.

'''Hidden::make('Category')'''

Any help at this stage would be simply marvellous. Thanks.

**Comment by AllTheUserNamesAreGone** [comment]
The format of the above post is not good and Github won't let me delete the post or it's content so reposting it here so it looks more readable.

Hi clutching at straws trying to get Filament to work and I came across Hidden which can be seen on the site like this:

```
use Filament\Forms\Components\Hidden;
 
Hidden::make('token')
```

I have tried creating a hidden TextInput field in order to transfer data from the form into the save method loading it with ```->default('ValueNeedToPassToSave')``` but needless to say that doesn't work when it's hidden.

Anyway I thought I'd try this ```Hidden::``` instead but I can not get a value into in order to get it to the form save() from my Resource file. I see the token 'Category' but is of course null. I have googled it looking for examples  but I find nothing and no example on how to use it on the Filament website either.

It says it needs a 'static' as the value but I have this in my form and have tried initialising with static variables, fn etc etc.

```Hidden::make('Category')```

Any help at this stage would be simply marvellous. Thanks.

**Reply by leandrocfe** [reply]
Why not customizing this data before saving?
https://filamentphp.com/docs/3.x/panels/resources/creating-records#customizing-data-before-saving

**Reply by AllTheUserNamesAreGone** [reply]
Thank you for your reply.

I'm trying to save it in a Hidden::Make() token because the data is read from the DB in the form so not available in the create or edit functions. I am trying to pass the data into the create and edit functions but creating a hidden input doesn't work, setting the default value doesn't work, I can't set it in mount() via form->fil(). Which is why I'm clutching at straws with a filament token.

I would've have though setting the token would be easy but so far all my attempts at getting a value you in there have failed.
